src/linkage.md

```diff
@@ -47,6 +47,16 @@ be ignored in favor of only building the artifacts specified by command line.
   linking Rust code into an existing non-Rust application
   because it will not have dynamic dependencies on other Rust code.
 
+  Note that, because the resulting static library contains all of the
+  dependencies, including the standard library, and also exports all public
+  symbols of them, linking the static library into an executable or shared
+  library will need special care. In case of a shared library the list of
+  exported symbols will have to be limited via e.g. a linker or symbol version
```

Comment:
I wouldn't say that they *have to* or that special care is *needed*.
It's more like "if you want to optimize the output in a similar way to rustc".

Reply:
In case of a shared library it seems more than an optimization as you're otherwise risking symbol conflicts, but yes.

How about softening it from the (current) "must" to "should"?